### ISSUE TITLE
Compiler error message CS1983 should describe that the type isn't TaskLike instead of saying it's not Task or Task<T>

### ISSUE THREAD

**Issue by davidfowl** [dotnet/roslyn#19616]
**Version Used**: 

C# 7

**Steps to Reproduce**:

```C#
using System;
using System.Threading.Tasks;

namespace ConsoleApp5
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");
        }

        public async MyTaskLike Foo()
        {
            await Task.Delay(1000);

            return 0;
        }
    }
}
```

**Expected Behavior**:

An error describing why `MyTaskLike` isn't being detected as returnable from an async method. E.g.

Couldn't find `MyTaskLike.CreateAsyncMethodBuilder`.

**Actual Behavior**:

The return type of an async method must be void, Task or Task<T>

PS: `ValueTask<T>` also works.

**Comment by davidfowl** [comment]
Stack overflow to the rescue http://stackoverflow.com/questions/42721979/in-c7-how-can-i-roll-my-own-task-like-type-to-use-with-async

**Comment by sharwell** [comment]
@davidfowl See also [Async Task Types in C#](https://github.com/dotnet/roslyn/blob/master/docs/features/task-types.md).

**Comment by sharwell** [comment]
💭 It's a bit hard to decide how I think this should go because of competing goals:

* The current messaging is simpler, and since `Task`/`Task<T>` are still the most generally-applicable types for applications to use, the current message is probably the shortest path to the right solution for almost all users.
* The current messaging does not help users who *do* want to use alternative types, and you're not the first person who's tried to figure out how to make it work.

**Comment by davidfowl** [comment]
Fair point but the compiler is inconsistent here. Take for example awating something arbitrary:

```C#
public async Task FooAsync()
{
    await new MyTaskLike();
}
```

**Comment by jcouv** [comment]
Yes, the diagnostics should be updated. This is tracked by https://github.com/dotnet/roslyn/issues/12621
Check it out to confirm it covers your feedback.

I will close the present issue as dupe.
